app/utils/shapes/heightAndWidthShape.ts

```diff
@@ -0,0 +1,95 @@
+import type TribontShape from 'mow-registry/models/tribont-shape';
+import {
+  DIMENSIONS,
+  dimensionToShapeDimension,
+  type shapeDimension,
+  type Shape,
+} from '.';
+import type Unit from 'mow-registry/models/unit';
+
+export default class HeightAndWidhtShape implements Shape {
+  height: shapeDimension;
+  width: shapeDimension;
+  shape: TribontShape;
+  constructor(
+    shape: TribontShape,
+    height: shapeDimension,
+    width: shapeDimension,
+  ) {
+    this.height = height;
+    this.width = width;
+    this.shape = shape;
+  }
+  toString() {
+    return `Hoogte: ${this.height.value} ${this.height.unit.symbol} Breedte: ${this.width.value} ${this.width.unit.symbol}`;
+  }
+  get unitMeasure() {
+    return this.height.unit;
+  }
+
+  get id() {
+    return this.shape.id as string;
+  }
+
+  static headers() {
+    return [
+      {
+        label: 'Hoogte',
+        value: 'height',
+      },
+      {
+        label: 'Breedte',
+        value: 'width',
+      },
+    ];
+  }
+
+  static async fromShape(shape: TribontShape) {
+    if (!shape.id) return;
+    const dimensions = await shape.dimensions;
+    const heightDimension = dimensions.find(
+      (dimension) => dimension.kind.id === DIMENSIONS.height,
+    );
+    const widthDimension = dimensions.find(
+      (dimension) => dimension.kind.id === DIMENSIONS.width,
+    );
+    if (!heightDimension || !widthDimension) return;
+    const height = await dimensionToShapeDimension(heightDimension, 'height');
+    const width = await dimensionToShapeDimension(widthDimension, 'width');
+    return new this(shape, height, width);
+  }
+
+  async convertToNewUnit(unit: Unit) {
+    this.height.unit = unit;
+    this.width.unit = unit;
+    this.height.dimension.set('unit', unit);
+    await this.height.dimension.save();
+    this.width.dimension.set('unit', unit);
+    await this.width.dimension.save();
+  }
+
+  async save() {
+    await this.height.dimension.save();
+    await this.width.dimension.save();
+    await this.shape.save();
+  }
+
+  async reset() {
+    this.height.dimension.rollbackAttributes();
```

Comment:
note that rollbackAttributes does _not_ reset relationships, can you confirm that's ok?